test/eth/TransactionManager.spec.js

```diff
@@ -208,4 +209,65 @@ describe('lifecycle hooks', () => {
     expect(biteHandlers.pending).toBeCalled();
     expect(biteHandlers.mined).toBeCalled();
   });
+
+  test('clear Tx when state is confirmed/finalized and older than 5 minutes', async () => {
+    const openId = uniqueId(open).toString();
+
+    const openHandlers = makeHandlers('open');
+    txMgr.listen(open, openHandlers);
+
+    // Add 10 minutes to the Tx timestamp
+    const myTx = txMgr._tracker.get(openId);
+    const minedDate = new Date(myTx._timeStampMined);
+    myTx._timeStampMined = minedDate.setMinutes(minedDate.getMinutes() + 10);
+
+    expect(txMgr._tracker._transactions).toHaveProperty(openId);
+
+    // after calling confirm, Tx state will become 'finalized' and be deleted from list.
+    await Promise.all([txMgr.confirm(open), mineBlocks(service)]);
+
+    expect(txMgr._tracker._transactions).not.toHaveProperty(openId);
+    expect(size(txMgr._tracker._listeners)).toEqual(
+      size(txMgr._tracker._transactions)
+    );
+  });
+
+  test('clear Tx when state is error and older than 5 minutes', async () => {
+    await Promise.all([txMgr.confirm(open), mineBlocks(service)]);
+
+    const lock = cdp.lockEth(0.01);
+    await Promise.all([lock, mineBlocks(service)]);
+
+    const draw = cdp.drawDai(1000);
+    const drawId = uniqueId(draw).toString();
+    const drawTx = txMgr._tracker.get(drawId);
+    const drawHandlers = makeHandlers('draw');
+
+    txMgr.listen(draw, drawHandlers);
+    expect(txMgr._tracker._transactions).toHaveProperty(drawId);
+
+    try {
+      await draw;
+    } catch (err) {
+      expect(drawTx.isError()).toBe(true);
+    }
+
+    // Add 10 minutes to the Tx timestamp
+    const minedDate = new Date(drawTx._timeStampMined);
+    drawTx._timeStampMined = minedDate.setMinutes(minedDate.getMinutes() + 10);
+
+    await mineBlocks(service);
+    expect(txMgr._tracker._transactions).not.toHaveProperty(drawId);
+  });
+
+  test('finalized Tx is set to correct state without without requiring a call to confirm()', async () => {
+    const openHandlers = makeHandlers('open');
+    txMgr.listen(open, openHandlers);
+    const openTx = txMgr._tracker.get(uniqueId(open));
+
+    await mineBlocks(service);
+
+    expect(openTx.isFinalized()).toBe(true);
```

Comment:
This does indeed fail under the "old" `isFinalized()` method, but idk if that's testable. Is this enough?

Reply:
Looks good to me.